cobbler/actions/buildiso/__init__.py

```diff
@@ -0,0 +1,251 @@
+                            pathlib.Path(self.api.settings().syslinux_dir).joinpath("modules/bios/"),
+                            pathlib.Path("/usr/lib/syslinux/"),
+                            pathlib.Path("/usr/lib/ISOLINUX/")]
+
+        # file_copy_success will be used to check for missing files
+        file_copy_success: Dict[str, bool] = {f: False for f in files_to_copy if f not in optional_files}
+        for syslinux_folder in syslinux_folders:
+            if syslinux_folder.exists():
+                for file_to_copy in files_to_copy:
+                    source_file = syslinux_folder.joinpath(file_to_copy)
+                    if source_file.exists():
+                        utils.copyfile(str(source_file), os.path.join(self.isolinuxdir, file_to_copy))
+                        file_copy_success[file_to_copy] = True
+
+        unsuccessful_copied_files = [k for k, v in file_copy_success.items() if not v]
+        if len(unsuccessful_copied_files) > 0:
+            self.logger.error("The following files were not found: \"%s\"", "\", \"".join(unsuccessful_copied_files))
+            raise FileNotFoundError("Required file(s) not found. Please check your syslinux installation")
+
+        self.logger.info("copying GRUB2 files")
+        bootloader_directory = pathlib.Path(self.api.settings().bootloaders_dir)
+        grub_efi = bootloader_directory.joinpath("grub", "grub.efi")
+        buildiso_directory = pathlib.Path(self.api.settings().buildisodir)
+        grub_target = buildiso_directory.joinpath("grub", "grub.efi")
+        if grub_efi.exists():
+            utils.copyfile(str(grub_efi), str(grub_target), symlink=True)
+        else:
+            self.logger.error("The following files were not found: \"%s\"", grub_efi)
+            raise FileNotFoundError("Required file(s) not found. Please check your GRUB 2 installation")
+
+    def __validate_buildisodir(self, buildisodir: str) -> str:
+        """
+        TODO
+
+        :param buildisodir:
+        :raises ValueError: In case the specified directory does not exist.
+        :raises TypeError: In case the specified argument is not of type str.
+        :return:
+        """
+        if not isinstance(buildisodir, str):
+            raise TypeError("buildisodir needs to be of type str!")
+        if not buildisodir:
+            buildisodir = self.api.settings().buildisodir
+        else:
+            if not os.path.isdir(buildisodir):
+                raise ValueError("The --tempdir specified is not a directory")
+
+            (_, buildisodir_tail) = os.path.split(os.path.normpath(buildisodir))
+            if buildisodir_tail != "buildiso":
+                buildisodir = os.path.join(buildisodir, "buildiso")
+
+        self.logger.info("Deleting and recreating the buildisodir at \"%s\"", buildisodir)
+        if os.path.exists(buildisodir):
+            shutil.rmtree(buildisodir)
+        os.makedirs(buildisodir)
```

Comment:
Yes I am aware. I guess the function name needs to be adjusted. And we REALLY do want to recreate those since we don't want to start with a non-empty buildisodir. I guess this mechanic could be improved but I guess this is a point for later...